services/webhook/deliver.go

```diff
@@ -170,18 +171,32 @@ func Deliver(ctx context.Context, t *webhook_model.HookTask) error {
 		}
 	}()
 
+	// OK We're now ready to attempt to deliver the task - we must double check that it
+	// has not been delivered in the meantime
+	updated, err := webhook_model.MarkTaskDelivered(ctx, t)
```

Comment:
Should we move these lines after line 194?

Reply:
No. We must do it as soon as the defer is in action.

Reply:
read the defer